addons/point_of_sale/models/pos_order.py

```diff
@@ -461,6 +469,9 @@ def write(self, vals):
         for order in self:
             if vals.get('state') and vals['state'] == 'paid' and order.name == '/':
                 vals['name'] = self._compute_order_name()
+            if vals.get('mobile'):
+                vals['mobile'] = self._phone_format(number=vals.get('mobile'),
+                        country=order.partner_id.country_id or self.env.company.country_id)
```

Comment:
maybe instead of that you can create a compute methods that depends on the partner.